vowpalwabbit/core/src/reductions/cb/cb_explore_adf_large_action_space.cc

```diff
@@ -25,28 +26,82 @@ namespace VW
 {
 namespace cb_explore_adf
 {
-// this is meant to be called with foreach_feature
-// for each feature it will multiply the weight that corresponds to the index with the corresponding gaussian element
-// If the returned values are summed the end resutlt is the dot product between the weight vector of the features of an
-// example with a vector of gaussian elements
 template <typename WeightsT>
-struct LazyGaussianDotProduct
+struct A_triplet_constructor
```

Comment:
Following the redsvd implementation for randomized truncated SVD from [here](https://github.com/ntessore/redsvd-h/blob/master/include/RedSVD/RedSVD-h). Keeping the equivalent matrix names is very useful, especially since this is still under construction and some steps/matrixes might be removed in the future